src/lib.rs

```diff
@@ -6,6 +6,21 @@ mod resources;
 mod server;
 mod web;
 
-pub async fn run() -> color_eyre::Result<Router> {
-    server::run().await
+pub use resources::DatabaseKind;
+pub use resources::DatabaseUrl;
+
+pub async fn run(config: Config) -> color_eyre::Result<Router> {
+    server::run(config).await
+}
+
+/// Configuration for the application.
+pub struct Config {
```

Comment:
**nit**: IMO, types should be declared before they're used; would prefer if `config` was defined before this

Reply:
I've moved the definition of Config so that it comes before it is used in `run`.